sdk/containerregistry/Azure.Containers.ContainerRegistry/src/ContainerRegistryClient.cs

```diff
@@ -51,17 +63,27 @@ protected ContainerRegistryClient()
         }
 
         /// <summary>
-        /// Ge the service endpoint for this client.
+        /// Gets the service endpoint for this client.
+        /// </summary>
+        public virtual Uri RegistryUri => _registryUri;
+
+        /// <summary>
+        /// Gets the name of this container registry.
         /// </summary>
-        public Uri Endpoint {  get { return _endpoint; } }
+        public virtual string Name => _registryName;
 
-        /// <summary> List repositories. </summary>
+        /// <summary>
+        /// Gets the login server name for this container registry.
+        /// </summary>
+        public virtual string LoginServer => _registryUri.Host;
+
+        /// <summary> List repositories in this registry. </summary>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
-        public virtual AsyncPageable<string> GetRepositoriesAsync(CancellationToken cancellationToken = default)
+        public virtual AsyncPageable<string> GetRepositoryNamesAsync(CancellationToken cancellationToken = default)
```

Comment:
All methods should at least document the `RequestFailedException`, along with any other exceptions e.g. `ArgumentException`, `ArgumentNullException` they might likely throw.